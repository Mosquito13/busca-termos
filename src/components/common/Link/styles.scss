@import '../../../sass/styles.scss';

.link {
  display: flex;
  background-color: transparent;
  color: currentColor;
  font-family: inherit;
  font-size: 1.6rem;
  border: none;
  border-bottom: 1px solid currentColor;
  cursor: pointer;
  transition: color 0.2s;

  &:hover {
    @include themify() {
      color: themed('color-contrast');
    }
  }

  &__icon {
    margin-left: 1rem;
  }
}
