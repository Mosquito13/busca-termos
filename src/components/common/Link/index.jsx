import PropTypes from 'prop-types';

import Icon, { RESPONSIVE } from '../Icon/index';

import './styles.scss';

const Link = ({ value, icon, onClick }) => {
  return (
    <button className="link" onClick={onClick}>
      <div className="link__text">{value}</div>
      {icon && (
        <div className="link__icon">
          <Icon size={RESPONSIVE}>{icon}</Icon>
        </div>
      )}
    </button>
  );
};

Link.propTypes = {
  /**
   * Texto exibido no link
   */
  value: PropTypes.string,
  /**
   * Icone exibido apos o link
   */
  icon: PropTypes.node,
  /**
   * Acao do link
   */
  onClick: PropTypes.func
};

export default Link;
