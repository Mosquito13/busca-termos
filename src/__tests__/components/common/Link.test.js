import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { FiAward } from 'react-icons/fi';

import Link from '../../../components/common/Link';

describe(`${Link.name}`, () => {
  describe('tests', () => {
    let wrapper, onClickSpy;

    beforeAll(() => {
      onClickSpy = jest.fn();

      wrapper = shallow(<Link value="Text" onClick={onClickSpy} />);
    });

    it('should render', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.link')).toHaveLength(1);
      expect(wrapper.text()).toContain('Text');
    });

    it('should fire onClick', () => {
      wrapper.find('.link').simulate('click');
      expect(onClickSpy).toBeCalledTimes(1);
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render without icon', () => {
      const tree = renderer.create(<Link value="Text" />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with icon', () => {
      const tree = renderer
        .create(<Link value="Text" icon={<FiAward />} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
