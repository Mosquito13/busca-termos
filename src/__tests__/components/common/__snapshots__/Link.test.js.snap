// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link snapshots should render with icon 1`] = `
<button
  className="link"
>
  <div
    className="link__text"
  >
    Text
  </div>
  <div
    className="link__icon"
  >
    <div
      className="icon"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="8"
          r="7"
        />
        <polyline
          points="8.21 13.89 7 23 12 20 17 23 15.79 13.88"
        />
      </svg>
    </div>
  </div>
</button>
`;

exports[`Link snapshots should render without icon 1`] = `
<button
  className="link"
>
  <div
    className="link__text"
  >
    Text
  </div>
</button>
`;
